tools/article_checker.py

```diff
@@ -0,0 +1,184 @@
+                if line.startswith("+"):
+                    # sanitize line
+                    line = line[1:]  # rm addition indicator
+                    line = line.strip()  # rm leading/trailing whitespace
+
+                    # filling the content for further check
+                    content += line + '\n'
+
+    return content
+
+
+def api_call(PROMPT, headers, endpoint):
+    """
+    Create an API call
+    """
+    data = {"Content": PROMPT}
+    try:
+        response = requests.post(endpoint, headers=headers, json=data)
+        response.raise_for_status()  # will raise an HTTPError if the HTTP request returned an unsuccessful status code
+        return response.json()
+    except requests.exceptions.RequestException as e:
+        print(f"Error making API call: {e}")
+    return None
+
+
+PROMPT = """
+Task: Conduct a thorough fact-check of the provided text by identifying each factual claim and verifying its accuracy against reliable web sources. For each claim, cross-reference the specific details such as numbers, dates, monetary values, and named entities with information available from credible websites.
+
+Present your findings only in a structured JSON format with each claim, the verification source, and the outcome of the verification.
+Output Format: 
+[
+  {
+    "claim": "[Exact factual statement from the text]",
+    "source": "[Direct URL or the name of the credible source where the verification information was found]",
+    "result": [true or false]
+  }
+]
+
+Example:
+Given Input Text: "In July 2011, BTC-e, a cryptocurrency exchange, experienced a security breach that resulted in the loss of around 4,500 BTC."
+Desired Output:
+[
+  {
+    "claim": "In July 2011, BTC-e experienced a security breach.",
+    "source": "https://bitcoinmagazine.com/business/btc-e-attacked-1343738085",
+    "result": false
+  }
+]
+
+Text for Verification: ```%s```
+"""
+
+
+@logging_decorator("Comment on PR")
+def create_beautiful_comment(pull_request, answer):
+    """
+    Create a comment on a Github PR.
+    """
+    try:
+        comment = "Here is a result of fact-checking:\n\n"
+        for claim in answer:
+            emoji = ":white_check_mark:" if claim["result"] else ":x:"
+            comment += f"`{claim['claim']}` {emoji}\n{claim['source']}\n\n"
+
+        print(comment)
+        # only post comment if running on Github Actions
+        if os.environ.get("GITHUB_ACTIONS") == "true":
+            pull_request.create_issue_comment(comment)
+    except Exception as e:
+        print(f"Error creating a comment on PR: {e}")
+
+
+def create_ugly_comment(pull_request, answer):
+    """
+    Create a comment on a Github PR.
+    """
+    comment = "Here is a result of fact-checking:\n\n"
+    comment += answer
+    print(comment)
+    # only post comment if running on Github Actions
+    if os.environ.get("GITHUB_ACTIONS") == "true":
+        pull_request.create_issue_comment(comment)
+
+
+def main():
```

Comment:
could you add a comment for the main that describes the functionality, detailed enough for GPT to be able to generate the working code